### DISCUSSION TITLE
Request: Ability to turn off Pull Requests

### DISCUSSION THREAD

**Discussion by joshmanders** [community/community#8907] [Pull Requests] [answered]
Right now you can [disable Issues](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/disabling-issues) if you decide you don't want to accept contributions or bug reports... but that doesn't stop people from submitting PR's in the event that you still don't to accept contributions or bug fixes.

Sometimes there's times we just want to share code and don't want the burden of maintaining it, triaging issues or pull requests, etc.

It would be nice to have the ability to disable pull requests just like you can disable issues.

More discussion and source of this request: https://twitter.com/adamwathan/status/1471151787017187331

**Comment by robertdrakedennis** [comment]
Also I feel some would argue that this is what gists are for. And that is simply not the case disabled pull requests could make it so much easier to share large or much more complex projects that normally wouldn't see the light of day due to having people make pull requests that the maintainer simply won't accept.

**Reply by denyago** [reply]
I thought too, but can you push a file tree to a gist? And have all those lovely features of Git repositories, like navigation, method references and a readme file in the front.

**Reply by nektro** [reply]
gists do not support folders no

**Reply by ADTC** [reply]
Saying "Use Gist instead of GitHub repo" is like saying "Use Notepad instead of your IDE". 😂

**Comment by thescientist13** [comment]
The Changelog had a great episode on this exact topic recently, definitely worth checking out; [_"Open source, not open contribution"_](https://changelog.com/podcast/433)
> _... His reason was to protect his mental health and the long term viability of the project. On this episode we talk with Ben about what that means, his thoughts on mental health and burnout in open source, ..._

Can't hurt to at least make it an option, right?

**Comment by nektro** [comment]
Also, Archiving and Interaction Limits are not a solution to this because I'd still like to allow issues/commits to be made, etc.

**Comment by yurivict** [comment]
So when it will be possible to disable pull requests?

**Comment by grahamperrin** [comment]
From <https://github.com/github-community/community/discussions/20346>: 

> Some projects, like FreeBSD, use GitGub only as a mirror, and they have their own ways to submit patches.
> 
> But GitHub doesn’t allow to disable pull requests, and users who submit pull requests anyway have their contributions delayed or lost because of this.
> 
> See the downstream bug report that can't be resolved without such GH feature: [bugs.freebsd.org/bugzilla/show_bug.cgi?id=265115](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=265115) …

Note that FreeBSD bug 265115 is specific to the `ports` repository. 

Please, do **not** disable pull requests (or GitLab merge requests) for the `doc` or `src` repos for FreeBSD.

**Comment by grahamperrin** [comment]
Cross-reference: https://github.com/dear-github/dear-github/issues/84

**Comment by Pharap** [comment]
To reiterate what I have said [elsewhere](https://github.com/dear-github/dear-github/issues/84#issuecomment-913617781)...

---

This would be a very welcome feature, but I would add the caveat that it should be possible to instead restrict PRs to being a maintainers only feature.

Often people make a PR that solves their immediate problem without considering the needs of others or the wider implications of their changes. Maintainers understand those needs and implications much better than drive-by contributors, so it makes sense to be able to restrict contributions to only maintainers.

A workflow that consists of users making feature requests and bug reports via issues and then maintainers deciding how best to implement those features or fix those bugs will often run more smoothly than users contributing a half-baked solution and maintainers having to gently explain why they aren't going to accept the PR without seeming like the villain of the story.

**Reply by ansgarbecker** [reply]
Very true. Several times I found myself in a discussion about a half baken solution which even introduced new issues. So restricting PRs would help to keep some level of software quality without wasting time with lost discussions.

**Comment by yurivict** [comment]
In FreeBSD maintainers don't even look at PRs, because there's another way to submit patches - Bugzilla.
Maintainers aren't notified of GitHub PRs and they likely get lost.

**Reply by grahamperrin** [reply]
Please see <https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=265115#c3>

**Comment by ghost** [comment]
Definitely seems to be a feature that has been requested plenty of times ([from as far back as 2016](https://github.com/dear-github/dear-github/issues/84)). There is plenty of use cases for this feature. Some of the main ones that resonate with me is repos that are just mirrors, where the repo has a different mechanism for contributing, or because your repo is a fork and PRs should be sent upstream instead of to the fork. 

Really hoping this is a feature that gets implemented.

**Comment by vinniefalco** [comment]
I don't want people submitting pull requests to my remote fork, and I do want people submitting pull requests to the source repository.

**Comment by stevenvachon** [comment]
I have a username repository (github.com/username/uesrname) for my GitHub profile's readme. No need for PRs.

**Comment by daztucker** [comment]
Reposting and expanding my comment from https://github.com/dear-github/dear-github/issues/84#issuecomment-1186134019:

I also would like to disable pull requests on a repo, but for a different reason to most mentioned here: security of selfhosted runners.  This is even more of a problem for us (OpenSSH) with the recent announcement that [selfhosted runners on private repos will soon have a per-minute monetary cost](https://resources.github.com/actions/2026-pricing-changes-for-github-actions/)!

We have a set of self-hosted runners that run tests on various platforms for which there are no hosted runners, and based on the docs we have them attached to a private repo that mirrors the public one.  I would like to be able to disable pull requests on that repo so I can make it and thus the test statuses public.

We set it up this way since the [action runner docs](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#self-hosted-runner-security) point out that the ability to create pull requests on a repo can be used to compromise the security of a runner attached to that repo.  [Secure use reference](https://docs.github.com/en/actions/reference/security/secure-use#hardening-for-self-hosted-runners) says:
> As a result, self-hosted runners should almost [never be used for public repositories](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions) on GitHub, because any user can open pull requests against the repository and compromise the environment.

There is an option to disable workflows in pull requests but I'm not sure if that it's sufficient, especially given that the runner docs don't mention it and have only a blanket warning against using selfhosted runners on public repos.

There are [Interaction Limits](https://docs.github.com/en/communities/moderating-comments-and-conversations/limiting-interactions-in-your-repository) which allows exactly this, but only for a limited time (up to 6 months) so for my use case, having a security control that fails open after some period is extremely problematic.

Please provide a way to *permanently* disable pull requests on a specific repo so that I can safely make this repo and its runner test results public.

**Comment by erbanku** [comment, marked as answer]
GitHub is finally implementing this.

> Features we’ll be shipping soon
> Repo-level pull request controls: Gives maintainers the option to limit pull request creation to collaborators or disable pull requests entirely. While the introduction of the pull request was fundamental to the growth of open source, maintainers should have the tools they need to manage their projects.
> Pull request deletion from the UI: Remove spam or abusive pull requests so repositories can stay more manageable.

https://github.blog/open-source/maintainers/welcome-to-the-eternal-september-of-open-source-heres-what-we-plan-to-do-for-maintainers/

**Reply by nektro** [reply]
release announced here #187038

**Reply by AverageHelper** [reply]
Looks to be available now! Time for me to disable PRs and Actions on my mirrors, now that I don't need an auto-pr-close workflow lol

https://github.blog/changelog/2026-02-13-new-repository-settings-for-configuring-pull-request-access/

**Reply by albertocavalcante** [reply]
It's great we finally have this.

https://x.com/jaredpalmer/status/2022395520623480970
